src/module/apps/characterImport/ArmorParser.js

```diff
@@ -1,13 +1,21 @@
-import { parseDescription, getArray, parseTechnology, createItemData } from "./BaseParserFunctions.js"
+import { parseDescription, getArray, parseTechnology, createItemData, formatAsSlug, genImportFlags } from "./BaseParserFunctions.js"
+import * as IconAssign from '../../apps/iconAssigner/iconAssign';
+import { SR5 } from "../../config";
 
 export class ArmorParser {
-   
-    parseArmors(chummerChar) {
+
+    async parseArmors(chummerChar, assignIcons) {
         const armors = getArray(chummerChar.armors.armor);
         const parsedArmors = [];
-        armors.forEach((chummerArmor) => {
+        const iconList = await IconAssign.getIconFiles();
+
+        await armors.forEach(async (chummerArmor) => {
             try {
                 const itemData = this.parseArmor(chummerArmor);
+
+                // Assign the icon if enabled
+                if (assignIcons) itemData.img = await IconAssign.iconAssign(itemData.system.importFlags, itemData.system, iconList);
```

Comment:
Can you add curly braces to all if`s even when they are one-liners?

Reply:
Fixed in ff9ff84494fbfa3f8a310949c69df9bebbe52d9a